### DISCUSSION TITLE
Flashlights should be a blunt melee weapon

### DISCUSSION THREAD

**Discussion by mygamingaccount** [FakeFishGames/Barotrauma#12266] [Ideas and suggestions]
Flashlights should have an attack action

**Comment by Smrtonosan** [comment]
I'd prefer not having the flashlight flail around while shooting a revolver, especially not for less damage than a wrench.

**Comment by ghost** [comment]
Why using 1 (one) revolver / melee weapon with flashlight?
Revolvers are performing better in pairs.
Same goes for melee.
If you've met some threat in really dark place - you're either in a suit with built-in lights, outgrew one-handed weaponry which is incompatible with underslung flashlight or face-to-face with your enemy (and could use a full-fledged melee weapon instead of extra light) already!

(also, flashlight still have
`Holdable slots="Any,RightHand,LeftHand,Head"`
in its tags so stick it in your ear if you're so desperate...)

**Reply by Smrtonosan** [reply]
You don't get better weapon until reaching zone 2.
You get wrecks and caves in zone 1.
Diving suit light barely illuminates your face, let alone anything else.
Flashlight mounted in the head slot doesn't face exactly where you are aiming, bad for wrecks.

![Joe biden - Copy (2)](https://github.com/Regalis11/Barotrauma/assets/104994469/6c3272d9-a66e-42c5-95ae-f68caf78f085)
Flashlight in hands always faces exactly where your crosshair is, in other words at what you are shooting, also you don't have to twist and turn your entire body just to get a light on the enemy, nor do you really need to face the enemy, useful for caves.

Yes, better guns with underslung flashlights are indeed better than the crappy starter revolver, please do tell me how you get better guns in zone 1 without:
manslaughter, whacky armoury heist, finding one in a wreck and going around in circle until you get the talent to craft one.

And I'm pretty sure that the person that started the discussion didn't intend to kill mudraptors with a flashlight,

**Reply by ghost** [reply]
> Yes, better guns with underslung flashlights are indeed better than the crappy starter revolver, please do tell me how you get better guns in zone 1 without:
> manslaughter, whacky armoury heist, finding one in a wreck and going around in circle until you get the talent to craft one.

SMG crafting unlock is a T2 Officer talent (3rd level-up) even in worst case scenario. Magazines availability may vary but it's not too far either. Neither item require any difficult-to-obtain materials to craft.
Assistants' Peer Learning stacks, making obtaining even better weaponry even faster - and there are always some expendable assistants around.

Also, explosive harpoons ARE a thing. Just if you do need to shoot something but can't / don't want to wait for real guns.
Underslung flashlight compatible!

**Reply by Smrtonosan** [reply]
My third eye has been opened and I can fully see the point of just rushing SMG production instead of purchasing stacks upon stacks of revolver rounds just to feed a single revolver, especially considering that the produced SMGs will have quality bonus' increased damage.

But I wouldn't discredit the revolver as not being a "real gun", as it can very effectively deal with the early enemies and fill in that gap before getting SMG/Shotgun production. Ammunition is widespread, only issue is you can't mount a flashlight, hence my original comment. Filling the gap before getting "real guns" the revolver can be effectively used to dispatch husks in wrecks and can also be useful for when the captain decides he's going to ram a 😊flock😊 of mudraptors, and you are forced to fight against 3 grouped up mudraptors in a flooded submarine and engaging in melee combat would practically be a death sentence.

I've been refusing to acknowledge the Harpoon Gun's existence to this point, as I don't acknowledge as a "real gun". But after trying out those explosive harpoons you mentioned in sub editor, it's basically a pocket grenade launcher, thanks for that, I never would have even considered using those.
